api/filters/replacement/replacement.go

```diff
@@ -57,18 +57,43 @@ func applyReplacement(nodes []*yaml.RNode, value *yaml.RNode, targets []*types.T
 
 func applyToNode(node *yaml.RNode, value *yaml.RNode, target *types.TargetSelector) error {
 	for _, fp := range target.FieldPaths {
-		t, err := node.Pipe(yaml.Lookup(strings.Split(fp, ".")...))
+		fieldPath := strings.Split(fp, ".")
+		var t *yaml.RNode
+		var err error
+		if target.Options != nil && target.Options.Create {
+			t, err = node.Pipe(yaml.LookupCreate(value.YNode().Kind, fieldPath...))
+		} else {
+			t, err = node.Pipe(yaml.Lookup(fieldPath...))
+		}
 		if err != nil {
 			return err
 		}
 		if t != nil {
-			// TODO (#3492): Use the field options to refine interpretation of the field
-			t.SetYNode(value.YNode())
+			setTargetValue(target.Options, t, value)
 		}
 	}
 	return nil
 }
 
+func setTargetValue(options *types.FieldOptions, t *yaml.RNode, value *yaml.RNode) {
+	if t.YNode().Kind == yaml.ScalarNode && value.YNode().Kind == yaml.ScalarNode &&
+		options != nil && options.Delimiter != "" {
+
+		tv := strings.Split(t.YNode().Value, options.Delimiter)
+		v := yaml.GetValue(value)
+		switch {
+		case options.Index < 0: // prefix
+			tv = append([]string{v}, tv...)
+		case options.Index > len(tv): // suffix
```

Comment:
should this be `>= len(tv)` ?

to be safe the test can cover prefix (adding), replacing 0, replacing len(v)-1, and suffix (adding)

we have no way to remove yet

Reply:
fixed!